src/QCQP/MOI_wrapper.jl

```diff
@@ -228,8 +228,14 @@ function _subs!(
         end
     end
     if !isempty(old_var)
+        to_old_map = Dict(zip(new_var, old_var))
```

Comment:
Good catch, can you add as comment that this is needed in case the order of `MP.variables(p.polynomial)` changes after substitution so we need to update the order in `p.variables` ?